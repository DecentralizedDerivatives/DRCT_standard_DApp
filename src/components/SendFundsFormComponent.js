import React from 'react';
import { Field } from 'redux-form';
import { InputGroup, InputGroupAddon } from 'reactstrap';
import InputText from './InputText';
import InputNumber from './InputNumber';
import Select from './Select.js';
import { factories } from '../factoryProvider';

export const SendFundsFormComponent = ({
  handleSubmit,
  onSubmit
}) => {
  return (
    <form autoComplete="off" onSubmit={handleSubmit(onSubmit)}>
      <InputGroup className="form-dropdown">
        <Field
          name="currency"
          label="Select Currency"
          component={Select}
          options={factories(true)}
        />
      </InputGroup>
      <InputGroup className="form-eth-input">
        <Field
          name="address"
          label="Swap Address"
          component={InputText} />
      </InputGroup>
      <InputGroup className="form-eth-input">
        <Field name="amount" label="Amount of Ether" component={InputNumber} />
        <InputGroupAddon addonType="append">ETH</InputGroupAddon>
      </InputGroup>

      <div className="form-submit-wrapper">
        <button type="submit" className="form-submit btn btn-primary">
          Send Funds
        </button>
      </div>
    </form>
  );
};

export default SendFundsFormComponent;
