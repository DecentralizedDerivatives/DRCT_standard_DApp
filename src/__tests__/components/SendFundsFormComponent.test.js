import SendFundsFormComponent from '../../components/SendFundsFormComponent';

describe('<SendFundsFormComponent />', () => {
  it('renders the component', () => {
    const handleSubmit = jest.fn();
    const wrapper = shallow(
      <SendFundsFormComponent store={initStore()} handleSubmit={handleSubmit} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
