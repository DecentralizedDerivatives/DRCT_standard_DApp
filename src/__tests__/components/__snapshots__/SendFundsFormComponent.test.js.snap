// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SendFundsFormComponent /> renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SendFundsFormComponent
    handleSubmit={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "autoComplete": "off",
      "children": Array [
        <InputGroup
          className="form-dropdown"
          tag="div"
        >
          <Field
            component={[Function]}
            label="Select Currency"
            name="currency"
            options={
              Object {
                "0x92217550aba5912ba7eb70978871daf7d6bcc16d": "BTC/USD",
                "0xf55e6ce774cec3817467aed5f5a5769f006658d0": "ETH/USD",
              }
            }
          />
        </InputGroup>,
        <InputGroup
          className="form-eth-input"
          tag="div"
        >
          <Field
            component={[Function]}
            label="Swap Address"
            name="address"
          />
        </InputGroup>,
        <InputGroup
          className="form-eth-input"
          tag="div"
        >
          <Field
            addonLabel="ETH"
            component={[Function]}
            label="Amount of Ether"
            name="amount"
          />
        </InputGroup>,
        <div
          className="form-submit-wrapper"
        >
          <button
            className="form-submit btn btn-primary"
            type="submit"
          >
            Send Funds
          </button>
        </div>,
      ],
      "onSubmit": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Field
            component={[Function]}
            label="Select Currency"
            name="currency"
            options={
              Object {
                "0x92217550aba5912ba7eb70978871daf7d6bcc16d": "BTC/USD",
                "0xf55e6ce774cec3817467aed5f5a5769f006658d0": "ETH/USD",
              }
            }
          />,
          "className": "form-dropdown",
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "label": "Select Currency",
            "name": "currency",
            "options": Object {
              "0x92217550aba5912ba7eb70978871daf7d6bcc16d": "BTC/USD",
              "0xf55e6ce774cec3817467aed5f5a5769f006658d0": "ETH/USD",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Field
            component={[Function]}
            label="Swap Address"
            name="address"
          />,
          "className": "form-eth-input",
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "label": "Swap Address",
            "name": "address",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Field
            addonLabel="ETH"
            component={[Function]}
            label="Amount of Ether"
            name="amount"
          />,
          "className": "form-eth-input",
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addonLabel": "ETH",
            "component": [Function],
            "label": "Amount of Ether",
            "name": "amount",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="form-submit btn btn-primary"
            type="submit"
          >
            Send Funds
          </button>,
          "className": "form-submit-wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Send Funds",
            "className": "form-submit btn btn-primary",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Send Funds",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "autoComplete": "off",
        "children": Array [
          <InputGroup
            className="form-dropdown"
            tag="div"
          >
            <Field
              component={[Function]}
              label="Select Currency"
              name="currency"
              options={
                Object {
                  "0x92217550aba5912ba7eb70978871daf7d6bcc16d": "BTC/USD",
                  "0xf55e6ce774cec3817467aed5f5a5769f006658d0": "ETH/USD",
                }
              }
            />
          </InputGroup>,
          <InputGroup
            className="form-eth-input"
            tag="div"
          >
            <Field
              component={[Function]}
              label="Swap Address"
              name="address"
            />
          </InputGroup>,
          <InputGroup
            className="form-eth-input"
            tag="div"
          >
            <Field
              addonLabel="ETH"
              component={[Function]}
              label="Amount of Ether"
              name="amount"
            />
          </InputGroup>,
          <div
            className="form-submit-wrapper"
          >
            <button
              className="form-submit btn btn-primary"
              type="submit"
            >
              Send Funds
            </button>
          </div>,
        ],
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Field
              component={[Function]}
              label="Select Currency"
              name="currency"
              options={
                Object {
                  "0x92217550aba5912ba7eb70978871daf7d6bcc16d": "BTC/USD",
                  "0xf55e6ce774cec3817467aed5f5a5769f006658d0": "ETH/USD",
                }
              }
            />,
            "className": "form-dropdown",
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "label": "Select Currency",
              "name": "currency",
              "options": Object {
                "0x92217550aba5912ba7eb70978871daf7d6bcc16d": "BTC/USD",
                "0xf55e6ce774cec3817467aed5f5a5769f006658d0": "ETH/USD",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Field
              component={[Function]}
              label="Swap Address"
              name="address"
            />,
            "className": "form-eth-input",
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "label": "Swap Address",
              "name": "address",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Field
              addonLabel="ETH"
              component={[Function]}
              label="Amount of Ether"
              name="amount"
            />,
            "className": "form-eth-input",
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "addonLabel": "ETH",
              "component": [Function],
              "label": "Amount of Ether",
              "name": "amount",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="form-submit btn btn-primary"
              type="submit"
            >
              Send Funds
            </button>,
            "className": "form-submit-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Send Funds",
              "className": "form-submit btn btn-primary",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Send Funds",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
